cmd/jaeger/internal/extension/jaegerquery/server_test.go

```diff
@@ -0,0 +1,308 @@
+			} else {
+				require.ErrorContains(t, err, tt.expectedErr)
+			}
+		})
+	}
+}
+
+func TestServerAddArchiveStorage(t *testing.T) {
+	host := componenttest.NewNopHost()
+
+	tests := []struct {
+		name           string
+		qSvcOpts       *querysvc.QueryServiceOptions
+		config         *Config
+		extension      component.Component
+		expectedOutput string
+		expectedErr    string
+	}{
+		{
+			name:           "Archive storage unset",
+			config:         &Config{},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			expectedOutput: `{"level":"info","msg":"Archive storage not configured"}` + "\n",
+			expectedErr:    "",
+		},
+		{
+			name: "Archive storage set",
+			config: &Config{
+				TraceStorageArchive: "random-value",
+			},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			expectedOutput: "",
+			expectedErr:    "cannot find archive storage factory: cannot find extension",
+		},
+		{
+			name: "Archive storage not supported",
+			config: &Config{
+				TraceStorageArchive: "badger",
+			},
+			qSvcOpts:       &querysvc.QueryServiceOptions{},
+			extension:      fakeStorageExt{},
+			expectedOutput: "Archive storage not supported by the factory",
+			expectedErr:    "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			logger, buf := testutils.NewLogger()
+			telemetrySettings := component.TelemetrySettings{
+				Logger: logger,
+			}
+			server := newServer(tt.config, telemetrySettings)
+			if tt.extension != nil {
+				host = storageHost{extension: tt.extension}
+			}
+			err := server.addArchiveStorage(tt.qSvcOpts, host)
+			if tt.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.ErrorContains(t, err, tt.expectedErr)
+			}
+
+			assert.Contains(t, buf.String(), tt.expectedOutput)
+		})
+	}
+}
+
+func TestServerShutdown(t *testing.T) {
+	tests := []struct {
+		name           string
+		hasQueryServer bool
+	}{
+		{
+			name:           "server.server nil",
+			hasQueryServer: false,
+		},
+		{
+			name:           "server.server not nil",
+			hasQueryServer: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			telemetrySettings := component.TelemetrySettings{
+				Logger: zap.NewNop(),
+			}
+			server := newServer(createDefaultConfig().(*Config), telemetrySettings)
+			assert.NotNil(t, server)
+
+			if tt.hasQueryServer {
+				serverOptions := &queryApp.QueryOptions{
+					HTTPHostPort: ":8080",
+					GRPCHostPort: ":8080",
+					QueryOptionsBase: queryApp.QueryOptionsBase{
+						Tenancy: tenancy.Options{
+							Enabled: true,
+						},
+					},
+				}
+				tenancyMgr := tenancy.NewManager(&serverOptions.Tenancy)
+				spanReader := &spanstoremocks.Reader{}
+				dependencyReader := &depsmocks.Reader{}
+				querySvc := querysvc.NewQueryService(spanReader, dependencyReader, querysvc.QueryServiceOptions{})
+				queryAppServer, err := queryApp.NewServer(zap.NewNop(), querySvc, nil,
+					serverOptions, tenancyMgr,
+					jtracer.NoOp())
```

Comment:
simplify (remove deps that don't make a difference)
```suggestion
				serverOptions := &queryApp.QueryOptions{
					HTTPHostPort: ":0",
					GRPCHostPort: ":0",
				}
				var tenancyMgr tenancy.Manager
				querySvc := querysvc.NewQueryService(nil, nil, querysvc.QueryServiceOptions{})
				queryAppServer, err := queryApp.NewServer(zap.NewNop(), querySvc, nil,
					serverOptions, &tenancyMgr, jtracer.NoOp())
```